src/components/Authors/Authors.tsx

```diff
@@ -0,0 +1,106 @@
+
+import React, { useEffect, useState } from 'react';
+import { format } from 'date-fns';
+import styled from '@emotion/styled';
+import axios from 'axios';
+
+/*
+Require: config.features.editOnRepo.editable = true
+We collect `Author` field, ref: https://stackoverflow.com/a/6755848/4540808
+*/
+
+export interface IAuthorsProps{
+  className: string;
+  path: string;
+  repoType: string;
+  repo: string;
+  contentRootPath: string;
+  locationPathname: string;
+}
+
+export interface IAuthorInfo{
+  username?: string; 
+  name: string;
+  commits: number;
+  url: string;
+}
+
+const initAuthorList :IAuthorInfo[] = [];
+
+const Authors = styled(({ className, path,  repoType, repo, contentRootPath, locationPathname}: IAuthorsProps) => {
+  const [authorList, setAuthorList] = useState(initAuthorList);
+  const [lastUpdate, setLastUpdate] = useState('');
+  if (repoType.toLowerCase() !== 'github') return <></>;
+  const contributorsGithubAPI = `https://api.github.com/repos/${repo}/commits?path=/${contentRootPath}/${path}`;
+
+  const fetchContributors = async () => {
+
+    const axiosConfig = {
+      headers: {
+        'Accept': 'application/vnd.github.v3+json'
+      }
+    }
+    const response = await axios.get(contributorsGithubAPI,axiosConfig);
+    console.log(response.data);
+
+    const authors: Record<string, IAuthorInfo> = {};
+    response.data.map( commit => {
+      // Get Github username or Email.
+      const author = commit?.author?.login || commit?.commit?.author.email;
+      if(!author) return;
+      console.log(commit);
+      if (author in authors){
+        authors[author].commits ++;
+      } else{
+        authors[author] = {
+          name: commit?.commit?.author.name,
+          commits: 1,
+          url: commit?.author?.html_url ||  commit?.html_url,
+        }
+      } 
+    });
+
+    const authorListTmp:IAuthorInfo[] = [];
+    for (const [key, value] of Object.entries(authors)) {
+      authorListTmp.push({
+        ...value,
+        username: key
+      });
+    }
+    authorListTmp.sort((a, b) => (a.commits < b.commits) ? 1 : -1);
+    setAuthorList([...authorListTmp]);
```

Comment:
For this block of code that we want to get a unique and sorted by commits count author, I suggest to use more declarative code, less mutate an object.

```ts

    // Map all commit to duplicated authors.
    const allAuthors: IAuthorInfo[] = response.data.map(commit => {
      return {
        // Get Github username or Email, sometime we can't get author from the API.
        username: commit?.author?.login || commit?.commit?.author.email,
        profileUrl: commit?.author?.html_url || commit?.html_url,
        commitsCount: 1, // default to 1
      };
    });

    // Create a unique author and increase commits count if an author has more than one commit.
    const uniqueAuthors = allAuthors.reduce((authors, currentAuthor) => {
      const usernameKey = currentAuthor.username;
      const uniqueAuthor = authors[usernameKey];
      if (uniqueAuthor) {
        uniqueAuthor.commitsCount++;
      } else {
        authors[usernameKey] = currentAuthor;
      }

      return authors;
    }, {} as Record<string, IAuthorInfo>);

    // Flatting to array of author and sort by commits number descending
    const sortedByCommitsCountAuthors = Object.entries(uniqueAuthors)
      .map(([, author]) => author)
      .sort((a, b) => (a.commitsCount < b.commitsCount) ? 1 : -1)

    // Set to React state
    setAuthorList([...sortedByCommitsCountAuthors]);
```

Reply:
I'll investigate on that.

Reply:
Nice suggestion, thank you so much.!